gc/base/MarkingScheme.cpp

```diff
@@ -263,7 +262,30 @@ MM_MarkingScheme::setMarkBitsInRange(MM_EnvironmentBase *env, void *heapBase, vo
  * Scanning
  ****************************************
  */
-	
+/**
+ * Private internal. Called exclusively from completeScan();
+ */
+uintptr_t
+MM_MarkingScheme::scanObject(MM_EnvironmentBase *env, omrobjectptr_t objectPtr)
+{
+	uintptr_t sizeToDo = UDATA_MAX;
+	GC_ObjectScannerState objectScannerState;
+	GC_ObjectScanner *objectScanner = _delegate.getObjectScanner(env, objectPtr, &objectScannerState, MM_MarkingDelegate::SCAN_REASON_PACKET, &sizeToDo);
+	if (NULL != objectScanner) {
+		bool isLeafSlot = false;
```

Comment:
Instead of using the local does it make not more sense to have a isCurrentSlotLeaf() on the scanner?